src/envelope.rs

```diff
@@ -6,8 +6,8 @@ pub(crate) struct Envelope {
     /// Who sent the message
     pub(crate) src: version::Dot,
 
-    /// When to deliver the message
-    pub(crate) deliver_at: Instant,
+    /// When to deliver the message. If `None`, the message is stuck without an ETA.
+    pub(crate) deliver_at: Option<Instant>,
```

Comment:
This is fine for this PR. I would probably lean towards a custom enum for clarity whenever this code is touched again.

Reply:
I agree, and will do that today. @mcches recommended I merge this asap to avoid rebases.

Reply:
this was fixed via https://github.com/tokio-rs/turmoil/pull/27